packages/dappmanager/src/modules/docker/utils.ts

```diff
@@ -68,3 +71,55 @@ export function getDockerTimeoutMax(
   }
   return timeout;
 }
+
+/**
+ * Returns the ports value returned by the docker api for a specific container.
+ * Example of ports: Dockerode.Ports[]
+ *   - "IP": "0.0.0.0" => for ipv4
+ *   - "IP": "::" => for ipv6
+ * ```
+ * "Ports": [
+      {
+        "PrivatePort": 80,
+        "Type": "tcp"
+      },
+      {
+        "IP": "0.0.0.0",
+        "PrivatePort": 8000,
+        "PublicPort": 2345,
+        "Type": "tcp"
+      },
+      {
+        "IP": "::",
+        "PrivatePort": 8000,
+        "PublicPort": 2345,
+        "Type": "tcp"
+      }
+    ]
+ * ```
+ */
+export function ensureUniquePortsFromDockerApi(
+  ports: Dockerode.Port[],
+  defaultPorts: PortMapping[] | undefined
+): PortMapping[] {
+  const portsFromApi = ports
+    .filter(port => port.IP && port.IP.trim() !== "::")
```

Comment:
Remove, don't make any assumptions on how the IP is rendered, de-duplication must be done per container port and protocol